### PR TITLE
[Rule Tuning] Some Tunings of several 8.9 rules

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
To ensure that all of the rules I pushed last release won't flood telemetry, I went through all of them and made some necessary changes. In two weeks I will take another look to see how they are doing.

`rules/_deprecated/command_and_control_connection_attempt_by_non_ssh_root_session.toml` is deprecated due to performance issues and its redundancy now that the reverse shell rules + suspicious network activity new_terms rule is out.

### CODE REVIEW

File: rules/linux/discovery_linux_modprobe_enumeration.toml [hunk 1]
```diff
@@ -47,9 +47,10 @@ tags = ["OS: Linux", "Use Case: Threat Detection", "Tactic: Discovery"]
 timestamp_override = "event.ingested"
 type = "eql"
 query = '''
-file where host.os.type == "linux" and event.action in ("opened-file", "read-file", "wrote-to-file") and
+file where host.os.type == "linux" and event.action == ("opened-file") and
```
Comment (Aegrah) [inline rules/linux/discovery_linux_modprobe_enumeration.toml:50]:
```suggestion
file where host.os.type == "linux" and event.action == "opened-file" and
```

File: rules/linux/discovery_linux_modprobe_enumeration.toml [hunk 2]
```diff
@@ -47,9 +47,10 @@ tags = ["OS: Linux", "Use Case: Threat Detection", "Tactic: Discovery"]
 timestamp_override = "event.ingested"
 type = "eql"
 query = '''
-file where host.os.type == "linux" and event.action in ("opened-file", "read-file", "wrote-to-file") and
+file where host.os.type == "linux" and event.action == "opened-file" and
 file.path : ("/etc/modprobe.conf", "/etc/modprobe.d", "/etc/modprobe.d/*") and not 
-(process.name in ("auditbeat", "kmod", "modprobe", "lsmod", "insmod", "modinfo", "rmmod") or process.title : ("*grep*") or process.parent.pid == 1)
+(process.name in ("auditbeat", "kmod", "modprobe", "lsmod", "insmod", "modinfo", "rmmod", "dpkg", "cp") or 
+ process.title : ("*grep*") or process.parent.pid == 1)
```
Comment (Aegrah) [inline rules/linux/discovery_linux_modprobe_enumeration.toml:53]:
```suggestion
 process.title : "*grep*" or process.parent.pid == 1)
```

File: rules/linux/discovery_linux_sysctl_enumeration.toml [hunk 3]
```diff
@@ -48,7 +48,8 @@ timestamp_override = "event.ingested"
 type = "eql"
 query = '''
 file where host.os.type == "linux" and event.action in ("opened-file", "read-file", "wrote-to-file") and
-file.path : ("/etc/sysctl.conf", "/etc/sysctl.d", "/etc/sysctl.d/*") and not process.name == "auditbeat"
+file.path : ("/etc/sysctl.conf", "/etc/sysctl.d", "/etc/sysctl.d/*") and 
+not process.name : ("auditbeat", "systemd-sysctl")
```
Comment (Aegrah) [inline rules/linux/discovery_linux_sysctl_enumeration.toml:52]:
```suggestion
not process.name in ("auditbeat", "systemd-sysctl")
```

File: rules/linux/persistence_init_d_file_creation.toml [hunk 4]
```diff
@@ -132,8 +132,8 @@ timestamp_override = "event.ingested"
 type = "new_terms"
 query = '''
 host.os.type :"linux" and event.action:("creation" or "file_create_event" or "rename" or "file_rename_event") and 
-file.path : /etc/init.d/* and not process.executable : ("/usr/bin/dpkg" or "/usr/bin/dockerd" or "/bin/rpm") and not 
-file.extension : "swp"
+file.path : /etc/init.d/* and not process.name : ("dpkg" or "dockerd" or "rpm" or "chef-client" or "apk" or "yum" or 
+"rpm" or "vmis-launcher" or "exe") and not file.extension : "swp"
```
Comment (Aegrah) [inline rules/linux/persistence_init_d_file_creation.toml:136]:
```suggestion
"rpm" or "vmis-launcher" or "exe") and not file.extension : ("swp" or "swx")
```

File: rules/linux/persistence_rc_script_creation.toml [hunk 5]
```diff
@@ -111,7 +111,7 @@ type = "new_terms"
 query = '''
 host.os.type : "linux" and event.category : "file" and 
 event.type : ("change" or "file_modify_event" or "creation" or "file_create_event") and
-file.path : "/etc/rc.local" and not file.extension : "swp"
+file.path : "/etc/rc.local" and not process.name : ("dockerd" or "yum" or "rpm" or "dpkg") and not file.extension : "swp"
```
Comment (Aegrah) [inline rules/linux/persistence_rc_script_creation.toml:114]:
```suggestion
file.path : "/etc/rc.local" and not process.name : ("dockerd" or "yum" or "rpm" or "dpkg") and not file.extension : ("swp" or "swx")
```

File: rules/linux/persistence_shared_object_creation.toml [hunk 6]
```diff
@@ -32,7 +32,7 @@ type = "new_terms"
 query = '''
 host.os.type:linux and event.action:(creation or file_create_event or file_rename_event or rename) and 
 file.path:(/dev/shm/* or /usr/lib/*) and file.extension:so and 
-process.name:(* and not (5 or dockerd or dpkg or rpm or snapd))
+process.name:(* and not (5 or dockerd or dpkg or rpm or snapd or exe or yum or vmis-launcher))
```
Comment (Aegrah) [inline rules/linux/persistence_shared_object_creation.toml:35]:
```suggestion
process.name: ( * and not ("5" or "dockerd" or "dpkg" or "rpm" or "snapd" or "exe" or "yum" or "vmis-launcher"))
```

**Review by w0rk3r — APPROVED** [review]
Nice! LGTM

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**